src/box/sql.c

```diff
@@ -1726,3 +1726,13 @@ sql_check_create(const char *name, uint32_t space_id, uint32_t func_id,
 	}
 	return sql_constraint_create(name, space_id, path, value);
 }
+
+#ifdef FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION
+int
+sql_fuzz(const char *sql, int bytes_count)
+{
+	struct Vdbe *stmt;
+	int rc = sql_stmt_compile(sql, bytes_count, NULL, &stmt, NULL);
```

Comment:
We need to finalize VDBE using `sqlVdbeFinalize()` if `sql_stmt_compile() was successful. I suggest to use something like:
```
if (sql_stmt_compile(sql, bytes_count, NULL, &stmt, NULL) != 0)
	return -1;
return sqlVdbeFinalize(stmt);
```

Reply:
Updated:

```diff
--- a/src/box/sql.c
+++ b/src/box/sql.c
@@ -1732,7 +1732,8 @@ int
 sql_fuzz(const char *sql, int bytes_count)
 {
        struct Vdbe *stmt;
-       int rc = sql_stmt_compile(sql, bytes_count, NULL, &stmt, NULL);
-       return rc;
+       if (sql_stmt_compile(sql, bytes_count, NULL, &stmt, NULL) != 0)
+               return -1;
+       return sqlVdbeFinalize(stmt);
 }
 #endif /* FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION */
```